src/test/run-make/json-errors/Makefile

```diff
@@ -0,0 +1,8 @@
+-include ../tools.mk
+
+all:
+	cp foo.rs $(TMPDIR)
+	cd $(TMPDIR)
+	$(RUSTC) -Z unstable-options --error-format=json foo.rs 2>foo.log || true
+	grep -q '{"message":"unresolved name `y`","code":{"code":"E0425","explanation":"\\nAn unresolved name was used. Example of erroneous codes.*"},"level":"error","span":{"file_name":"foo.rs","byte_start":523,"byte_end":524,"line_start":14,"line_end":14,"column_start":18,"column_end":19},"children":\[\]}' foo.log
```

Comment:
Another question I have is whether it is worth at all to include explanations for each error code too? It should inflate size of output greatly, especially if we do that for each error.